src/State/Scope.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace Sentry\State;
+
+use Sentry\Breadcrumbs\Breadcrumb;
+use Sentry\Context\Context;
+use Sentry\Context\TagsContext;
+use Sentry\Context\UserContext;
+use Sentry\Event;
+use Sentry\Interfaces\Severity;
+
+final class Scope
+{
+    /**
+     * Array holding all breadcrumbs.
+     *
+     * @var array
+     */
+    private $breadcrumbs = [];
+
+    /**
+     * @var null|UserContext
+     */
+    private $user = null;
+
+    /**
+     * Array holding all tags.
+     *
+     * @var null|TagsContext
+     */
+    private $tags = null;
+
+    /**
+     * Array holding all extra.
+     *
+     * @var null|Context
+     */
+    private $extra = null;
+
+    /**
+     * Array holding all fingerprints. This is used to group events together in Sentry.
+     *
+     * @var string[]
+     */
+    private $fingerprint = [];
+
+    /**
+     * @var null|Severity
+     */
+    private $level = null;
+
+    /**
+     * Array of eventProcessors. Closure receiving the event, they should return an event or null.
+     *
+     * @var array
+     */
+    private $eventProcessors;
```

Comment:
We can probably recycle the middleware here, in the future.

Reply:
Yes, kind of.